app/models/amendment.rb

```diff
@@ -33,15 +33,15 @@ def self.amend_claim(claim, claim_attributes, amendment_attributes)
     Claim.transaction do
       claim.assign_attributes(claim_attributes)
 
-      unless claim.save(context: :submit)
+      unless claim.save(context: [:submit, :amendment])
         amendment.valid?
         amendment.errors.merge!(claim.errors)
         amendment.errors.delete(:claim_changes)
         raise ActiveRecord::Rollback
       end
 
-      amendment.claim_changes = claim.previous_changes
-        .slice(*Claim::AMENDABLE_ATTRIBUTES)
+      amendment.claim_changes = claim.previous_changes.merge(claim.eligibility.previous_changes)
```

Comment:
I feel like this chain of hash munging has got long enough to warrant extracting it to a method to try and make things more easily understood.